### PR TITLE
Make `getPropertyOptions` public

### LINKED ISSUE #1945: Should `ReactiveElement.getPropertyOptions` be public?
In writing Apollo Elements I've had opportunity to do some hacking into `ReactiveElement`'s lifecycle. Specifically, I wrote a TypeScript decorator which forwards fields from a controller to the host and vice-versa.

```ts
function defineOnReactiveElement<T extends ReactiveElement & ApolloControllerHost>(
  proto: T,
  name: string & keyof T,
  opts: DefineOptions
): void {
  // Run our property effects
  defineOnHTMLElement(proto, name, {
    ...opts,
    // In addition to any user-defined side-effects,
    // also notify the ReactiveElement lifecycle
    onSet(this: T, x: unknown) {
      const old = this[name];
      opts?.onSet?.call?.(this, x);
      this.requestUpdate(name, old);
    },
  });
  // And also run ReactiveElement's property effects
  const Class = proto.constructor as typeof ReactiveElement;
  // @ts-expect-error: i know it's protected
  Class.createProperty(name, Class.getPropertyOptions(name));
}
```

In doing so, I end up calling `createProperty` (which is public) with `getPropertyOptions` (which is protected). In order to lubricate this workflow for TypeScript users, should `getPropertyOptions` be public?

### PR DESCRIPTION
**Pull request by bennypowers**
fixes #1945

### CODE REVIEW

**Comment by bennypowers** [comment]
see also https://github.com/material-components/material-web/blob/master/packages/base/aria-property.ts#L16

**Comment by bennypowers** [comment]
What are the odds we'll see this in 2.2?

**Comment by aomarks** [comment]
Sorry for the delay merging this. Would you mind merging with master, and also adding a changeset (with `npx changeset`). I think this should be a minor patch to reactive-element.

**Comment by bennypowers** [comment]
with pleasure

**Comment by bennypowers** [comment]
@aomarks PTAL

File: .changeset/clever-fans-decide.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@lit/reactive-element': patch
```
[the remark this answers is not in the document]
Comment (bennypowers) [reply .changeset/clever-fans-decide.md]: ✅

**Review by aomarks — APPROVED** [review]

**Pull request merged**